iP/src/duke/Parser.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+import java.time.LocalDate;
+import java.util.HashMap;
+
+/**
+ * Parse the user input is useable form
+ */
+public class Parser {
+    /**
+     * parse and get the instruction type of user input
+     * @param user_input input give from user
+     * @return instruction type of user input
+     */
+    static String parseInstruction(String user_input) {
+        String instructionType = user_input.split(" ")[0];
+        return instructionType;
+    }
+
+    /**
+     * parse instruction for marking tasks done and return index of task
+     * @param user_input input give from user
+     * @return index of task which will be marked as done
+     */
+    static int parseMarkDoneInstr(String user_input) {
+        int index = Integer.parseInt(user_input.split(" ")[1]) - 1;
+        return index;
+    }
+
+    /**
+     * parse instruction for deleting task
+     * @param user_input input give from user
+     * @return index of task which will be deleted
+     */
+    static int parseDeleteInstr(String user_input) {
+        int index = Integer.parseInt(user_input.split(" ")[1]) - 1;
+        return index;
+    }
+
+    /**
+     * parse instruction for adding todo
+     * @param user_input input give from user
+     * @return description of todo
+     */
+    static String parseAddTodoInstr(String user_input) throws DukeException {
+        // check if input is valid
+        if (user_input.split(" ", 2).length == 1) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        // get data
+        String description = user_input.split(" ", 2)[1];
+        return description;
+    }
+
+    /**
+     * parse instruction for adding deadline
+     * @param user_input input give from user
+     * @return hashMap containing description and deadline of task
+     */
+    static HashMap<String, Object> parseAddDeadlineInstr(String user_input) throws DukeException {
+        // check if input is valid
+        if (user_input.split(" ", 2).length == 1) {
+            throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+        }
+        String task = user_input.split(" ", 2)[1];
+        if (task.split(" /by ", 2).length < 2) {
+            throw new DukeException("☹ OOPS!!! The description and time is required for deadline");
+        }
+        // get data
+        String description = task.split(" /by ", 2)[0];
+        String time = task.split(" /by ")[1];
+        LocalDate l_time = LocalDate.parse(time);
+        HashMap<String, Object> parsedData = new HashMap<String, Object>(){
+            { put("description", description); put("time", l_time); }
+        };
+        return parsedData;
+    }
```

Comment:
This block is generally abit chaotic, perhaps add in line breaks to separate different logical units?